Doc/whatsnew/3.12.rst

```diff
@@ -575,6 +575,12 @@ New Features
   :c:func:`PyDict_AddWatch` and related APIs to be called whenever a dictionary
   is modified. This is intended for use by optimizing interpreters, JIT
   compilers, or debuggers.
+  (Contributed by Carl Meyer in :gh:`91052`.)
+
+* Added :c:func:`PyType_AddWatcher` and :c:func:`PyType_Watch` API to register
```

Comment:
what about `PyType_ClearWatcher`?

Reply:
I don't think "What's New" requires a detailed listing of every API function added; it's sufficient to outline the feature and link to reference docs where all the functions are clearly listed.